connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionSearch.java

```diff
@@ -71,9 +71,13 @@ public void query(Consumer<List<ProcessDefinition>> resultHandler) {
         paginationIndex = newPaginationIdx;
       }
 
-      processDefinitions.addAll(processDefinitionResult.getItems());
+      var items = processDefinitionResult.getItems();
+      if (items != null) {
+        processDefinitions.addAll(items);
+      }
 
-    } while (processDefinitionResult.getItems().size() > 0);
+    } while (processDefinitionResult.getItems() != null
+        && processDefinitionResult.getItems().size() > 0);
```

Comment:
🙃 Just a small note: maybe instead of `processDefinitionResult.getItems().size() > 0` we could also use `!processDefinitionResult.getItems().isEmpty()` if you would like to.